### PR TITLE
Add dilated convolution explanation to convolution arithmetic tutorial

### LINKED ISSUE #4962: Update convolution tutorial about dilation
http://deeplearning.net/software/theano_versions/dev/tutorial/conv_arithmetic.html

It is documented only in the library doc. Not very user friendly: http://deeplearning.net/software/theano_versions/dev/library/tensor/nnet/conv.html

### PR DESCRIPTION
**Pull request by vdumoulin**
Fixes #4962. @slefrancois

### CODE REVIEW

File: doc/tutorial/conv_arithmetic.txt [hunk 1]
```diff
@@ -878,8 +878,69 @@ Here is an example for :math:`i = 6`, :math:`k = 3`, :math:`s = 2` and :math:`p
...
     :figclass: align-center
 
+Miscellaneous convolutions
+==========================
+
+Dilated convolutions
+--------------------
+
+Those familiar with the deep learning literature may have noticed the term
+"dilated convolutions" (or *convolutions à trous*) appear in recent papers. Here
```
Comment (nouiz) [inline doc/tutorial/conv_arithmetic.txt]: There is also the term "atrous convolution", the english version. Add it somewhere to help people find it. They could grep for that.

Comment (vdumoulin) [inline doc/tutorial/conv_arithmetic.txt]: Done. 59e3cce

**Comment by slefrancois** [comment]
@vdumoulin, can you add a single-frame PDF version of the new figure? Same filename and location except the extension. Sphinx can't insert animated gifs in the PDF version of the doc.

File: doc/tutorial/conv_arithmetic.txt [hunk 2]
```diff
@@ -878,8 +878,70 @@ Here is an example for :math:`i = 6`, :math:`k = 3`, :math:`s = 2` and :math:`p
...
+.. admonition:: Relationship 14
+
+    For any :math:`i`, :math:`k`, :math:`p` and :math:`s`, and for a dilation
+    rate :math:`d`,
+
+    .. math::
+
+        o = \left\lfloor \frac{i + 2p - k - (k - 1)(d - 1)}{s} \right\rfloor + 1.
+
+    This translates to the following Theano code:
```
Comment (slefrancois) [inline doc/tutorial/conv_arithmetic.txt]: using the parameter filter_dilation

Comment (vdumoulin) [inline doc/tutorial/conv_arithmetic.txt]: Done. 039ce7d

**Comment by vdumoulin** [comment]
@slefrancois RE: single-frame PDF version: done. 1c982a4

**Comment by slefrancois** [comment]
Will merge on tests pass.

**Pull request merged**